### PR TITLE
Fix nightly CD for python docker image releases

### PR DESCRIPTION
**Pull request by mseth10**
## Description ##
This PR makes multiple changes to fix the nightly CD pipeline for python docker image releases.
- Change push target from dockerhub to ecr public repository
- Update all base images for docker build from ubuntu16 to ubuntu18
- Use deadsnakes python package
- Remove unused mxnet_lib/dynamic pipeline

### CODE REVIEW

**Review by leezu — COMMENTED** [review]
As part of this work, can we reconsider the publication strategy? Currently nightly builds are pushed to dockerhub, which is not compatible with ASF policy. Instead, can you configure the job to only push on approved release tags?

File: cd/python/docker/Dockerfile [hunk 1]
```diff
@@ -27,6 +27,7 @@ ARG PYTHON=python3
 ARG PIP=pip3
 ARG PYTHON_VERSION=3.7.9
 RUN apt-get update && \
+    apt-get install -y wget ${PYTHON}-dev gcc && \
```
Comment (josephevans) [inline cd/python/docker/Dockerfile:30]: You should be able to remove the "${PYTHON}-dev", since we're building from source.

**Comment by mseth10** [comment]
> As part of this work, can we reconsider the publication strategy? Currently nightly builds are pushed to dockerhub, which is not compatible with ASF policy. Instead, can you configure the job to only push on approved release tags?

As part of this PR, the docker push target has been moved to a public ECR repository.

File: cd/utils/mxnet_base_image.sh [hunk 2]
```diff
@@ -31,7 +31,7 @@ case ${mxnet_variant} in
     echo "nvidia/cuda:11.0-cudnn8-runtime-ubuntu16.04"
     ;;
     cu112*)
-    echo "nvidia/cuda:11.2-cudnn8-runtime-ubuntu16.04"
+    echo "nvidia/cuda:11.2.1-cudnn8-runtime-ubuntu16.04"
```
Comment (sandeep-krishnamurthy) [inline cd/utils/mxnet_base_image.sh:34]: ubuntu 16.04 is EOL by April, 2021 (https://ubuntu.com/about/release-cycle), Can we please move to ubuntu 18.04 or 20.04?

File: cd/utils/mxnet_base_image.sh [hunk 3]
```diff
@@ -22,22 +22,22 @@ mxnet_variant=${1:?"Please specify the mxnet variant as the first parameter"}
 
 case ${mxnet_variant} in
     cu101*)
-    echo "nvidia/cuda:10.1-cudnn7-runtime-ubuntu16.04"
+    echo "nvidia/cuda:10.1-cudnn7-runtime-ubuntu18.04"
```
Comment (access2rohit) [inline cd/utils/mxnet_base_image.sh:25]: Were you able to test these changes that it works with ubuntu18.04 ?

Comment (Zha0q1) [reply cd/utils/mxnet_base_image.sh:25]: I think the base image is specified in the docker files? here we might be just echoing a string.

Comment (Zha0q1) [reply cd/utils/mxnet_base_image.sh:25]: are those images only for the docker image pipelines?

File: cd/utils/mxnet_base_image.sh (hunk 2 above)
Comment (mseth10) [reply cd/utils/mxnet_base_image.sh]: Moved to ubuntu 18.04 for all docker base images

File: cd/utils/mxnet_base_image.sh (hunk 3 above)
Comment (mseth10) [reply cd/utils/mxnet_base_image.sh:25]: @access2rohit yes I tested it.

@Zha0q1 yes these base images are only for building docker images that get published.

**Review by access2rohit — APPROVED** [review]
LGTM!

**Review by Zha0q1 — APPROVED** [review]

**Pull request merged**